tests/integration/test_ols.py

```diff
@@ -249,3 +249,56 @@ def test_post_question_on_noyaml_response_type() -> None:
                         "The following response was generated without access to reference content:"
                         in response.json()["response"]
                     )
+
+
+def test_post_query_with_question_filter_response_type() -> None:
+    """Check the REST API /v1/query with POST HTTP method with question filter."""
+    config.init_empty_config()
+    config.ols_config.reference_content = ReferenceContent(None)
+    config.ols_config.reference_content.product_docs_index_path = "./invalid_dir"
+    config.ols_config.reference_content.product_docs_index_id = "product"
+    question_filters = [
+        Filter(
+            {
+                "name": "test",
+                "regular_expression": r"(?:\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})",
+                "replace_with_string": "redacted_ip",
+            }
```

Comment:
also a matter of taste, `redacted_ip` could be XXX.XXX.XXX.XXX

Reply:
user can choose what they want to replace with. This is only a test.